exercises/exercise5/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Node struct {
+	Value int
+	Next  *Node
+}
+
+func Prepend(head **Node, x int) {
+
+	*head = &Node{Value: x, Next: *head}
+}
+
+func Append(head **Node, x int) {
+
+	current := *head
+	for current.Next != nil {
+		current = current.Next
+	}
+
+	Prepend(&current, x)
+}
+
+func Insert(head **Node, x int) {
+
+	if x < (*head).Value {
```

Comment:
This implementation is fairly long and complicated, you can handle everything as a single case (double-pointers allow that)
I won't write an implementation here to let you have some fun with it, but trust me, it's a two-liner (assuming the for-loop takes a single line), and it would cover all possible scenarios